packages/xo-web/src/xo-app/settings/audit/index.js

```diff
@@ -211,35 +222,118 @@ const COLUMNS = [
       event === 'apiCall' ? data.method : event,
   },
   {
-    itemRenderer: ({ time }) => (
-      <FormattedDate
-        day='numeric'
-        hour='2-digit'
-        minute='2-digit'
-        month='short'
-        second='2-digit'
-        value={new Date(time)}
-        year='numeric'
-      />
-    ),
+    itemRenderer: ({ time }) =>
+      time !== undefined && (
+        <FormattedDate
+          day='numeric'
+          hour='2-digit'
+          minute='2-digit'
+          month='short'
+          second='2-digit'
+          value={new Date(time)}
+          year='numeric'
+        />
+      ),
     name: _('date'),
-    sortCriteria: 'time',
+    sortCriteria: ({ fakeTime, time = fakeTime }) => time,
     sortOrder: 'desc',
   },
+  {
+    itemRenderer: ({ id }, { checkedRecords, missingRecord }) => {
+      if (missingRecord === undefined && isEmpty(checkedRecords)) {
+        return
+      }
+
+      if (missingRecord === id) {
+        return (
+          <span className='text-danger'>
+            <Icon icon='error' /> {_('missing')}
+          </span>
+        )
+      }
+
+      if (checkedRecords[id]) {
+        return (
+          <span className='text-success'>
+            <Icon icon='success' /> {_('verified')}
+          </span>
+        )
+      }
+
+      if (checkedRecords[id] === false) {
+        return (
+          <span className='text-danger'>
+            <Icon icon='error' /> {_('altered')}
+          </span>
+        )
+      }
+    },
+    name: _('integrity'),
+  },
 ]
 
 export default decorate([
   provideState({
     initialState: () => ({
-      records: undefined,
+      _records: undefined,
+      checkedRecords: {},
+      goTo: undefined,
+      missingRecord: undefined,
     }),
     effects: {
       initialize({ fetchRecords }) {
         return fetchRecords()
       },
       async fetchRecords() {
-        this.state.records = await fetchAuditRecords()
+        this.state._records = await fetchAuditRecords()
       },
+      handleRef(_, ref) {
+        this.state.goTo = ref.goTo.bind(ref)
+      },
+      handleCheck(_, oldest, newest, error) {
+        const { state } = this
+        const checkedRecords = { ...state.checkedRecords }
+
+        if (error !== undefined) {
+          const { id } = error.data
+          if (missingAuditRecord.is(error)) {
+            state.missingRecord = id
+          } else {
+            checkedRecords[id] = false
```

Comment:
It will be true in the loop below, no?

Reply:
No, it will not be checked because it's the oldest see https://github.com/vatesfr/xen-orchestra/pull/4879/files#diff-458afd3b9cc8e211fa5bd0a0671aed78R305